Adding windows agent user new logic (#4340)
DataDog/documentation: content/en/agent/faq/windows-agent-ddagent-user.md

---
title: ddagentuser with the Windows Agent 
kind: faq
disable_toc: true
---

> **Starting with release `6.11`, the Windows Agent main components (i.e. Core, APM/Trace, and Process) run under the `ddagentuser` account, created at install time, instead of running on prior version under the `LOCAL_SYSTEM` account.**

Reviewer: Same (Process not running as ddagentuser, still runs as LOCAL_SYSTEM)

The user `ddagentuser` is created at install time for the Datadog Windows Agent. When installed on an Active Directory server, the username and password must be provided to the installer. The new user is a non-privileged user. It gains the following rights during installation:

* It can start and stop the APM and Process Agent
* It becomes a member of the “Performance Monitor Users” group

> * Necessary to access WMI information

Reviewer: These two lines (`necessary....`) might be indented?  They indicate _why_ we add the user to the Performance Monitor Users group

---
title: ddagentuser with the Windows Agent
kind: faq
disable_toc: true
---

> **Starting with release `6.11`, the Windows Agent main components (i.e. Core, and APM/Trace) run under the `ddagentuser` account, created at install time, instead of running on prior version under the `LOCAL_SYSTEM` account.**

Reviewer 2: ```suggestion
**Starting with release `6.11.0`, the Core and APM/Trace components of the Windows Agent run under the `ddagentuser` account, created at install time, instead of running on prior versions under the `LOCAL_SYSTEM` account. If enabled, the Live Process component still runs under `LOCAL_SYSTEM`. **
```

> **Starting with release `6.11.0`, the Core and APM/Trace components of the Windows Agent run under the `ddagentuser` account, created at install time, instead of running on prior versions under the `LOCAL_SYSTEM` account. If enabled, the Live Process component still runs under `LOCAL_SYSTEM`.**

Reviewer: same comment as above.

> **Starting with release `6.11.0`, the Core and APM/Trace components of the Windows Agent run under the `ddagentuser` account, created at install time, instead of running on prior versions under the `LOCAL_SYSTEM` account. If enabled, the Live Process component still runs under `LOCAL_SYSTEM`. **

Reviewer 2: ```suggestion
**Starting with release `6.11.0`, the Core and APM/Trace components of the Windows Agent run under the `ddagentuser` account, created at install time, instead of running on prior versions under the `LOCAL_SYSTEM` account. If enabled, the Live Process component still runs under `LOCAL_SYSTEM`.**
```

* It has remote login disabled
* It has network login disabled

## Installation
### Installation with Group Policy

The installer changes the local group policy to allow the newly created user account, `ddagentuser`, to **run as a service**.  If the domain group policy disallows that, then the installation setting is overridden, and the domain group policy has to be updated to allow the user to run as a service.

### Installation in a Domain environment

The Agent installer creates the `ddagentuser` at install time, and then registers the service (with the randomly generated password). The user is created as a local user, even in a domain environment, so that every machine on which the Agent is installed has a unique user and password.

The exception is on domain controllers (primary and backup). There is no notion of a local user on a domain controller. Therefore, the created user becomes a domain user rather than a local one.

To support this environment, the Agent installer requires that the administrator provides a username and password under which the Agent run. The username and password are provided as properties on the installation command line, i.e.

```shell

> 	Msiexec /i ddagent.msi DDAGENTUSER_NAME=<DOMAIN>\<USERNMAE> DDAGENTUSER_PASSWORD=<PASSWORD>

Reviewer: typo `USERNMAE` (probably copy and pasted from my doc :)

For example, if the directory check is monitoring a directory that has specific access rights, e.g. allowing only members of the Administrators group read access, then the existing Agent currently can monitor that directory successfully since `LOCAL_SYSTEM` has administrator rights. Upon upgrading, the administrator must add `ddagentuser` to the access control list for that directory in order for the directory check to function.

**Note**: The same considerations apply to the log files that may be monitored by the Logs Collection features of the Agent.

### JMX-based integrations

The change to `ddagentuser` affects your JMX-based integrations if the Agent’s JMXFetch is configured to connect to the monitored JVMs through the Attach API, i.e. if:

1. you’re using a JMX-based integration, i.e. one of the following integrations:
   * [ActiveMQ][1]
   * [ActiveMQ_XML][2]
   * [Cassandra][3]
   * [JMX][4]
   * [Presto][5]
   * [Solr][6]
   * [Tomcat][7]
   * [Kafka][8]

2. **AND** you’ve configured the integration with the `process_name_regex` setting instead of the `host` and `port` settings

> If you’re using the Attach API, the change in user context means that the Agent’s JMXFetch is only be able to connect to the JVMs that also run under the `ddagentuser` user context. In most cases, it's recommended that you switch JMXFetch by enabling JMX Remote on your target JVMs and configuring your JMX integrations using `host` and `port`. For more information, refer to the [JMX documentation][4].

Reviewer 2: sorry this is my bad, I wrote a sentence that doesn't mean much

```suggestion
If you’re using the Attach API, the change in user context means that the Agent’s JMXFetch is only be able to connect to the JVMs that also run under the `ddagentuser` user context. In most cases, it's recommended that you switch JMXFetch to using JMX Remote by enabling JMX Remote on your target JVMs and configuring your JMX integrations using `host` and `port`. For more information, refer to the [JMX documentation][4].
```
